### PR TITLE
Implemented claim

### PR DESCRIPTION
**Pull request by MartinSchmidt**
- Implemented claim logic.
- Added claim table
- Minor refactoring

### CODE REVIEW

File: src/ProjectOrigin.WalletSystem.IntegrationTests/FlowTests/ClaimTests.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        var prodCommitment = new SecretCommitmentInfo(issuedAmount);
+        var position = 1;
+        var issuedEvent = await _registryFixture.IssueCertificate(type, prodCommitment, senderDepositEndpoint.PublicKey.Derive(position).GetPublicKey());
+        await _dbFixture.InsertSlice(senderDepositEndpoint, position, issuedEvent, prodCommitment);
+        return issuedEvent.CertificateId;
+    }
+
+    [Fact]
+    public async Task Claim150_FromTwoLargerSlices_Success()
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/FlowTests/ClaimTests.cs:48]: Why only claim 150 when consumption is 300?

File: src/ProjectOrigin.WalletSystem.IntegrationTests/RegistryProcessBuilder/ClaimTests.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        var consPublicKey = depositEndpoint.PublicKey.Derive(consSlice.DepositEndpointPosition);
+
+        // Act
+        await _processBuilder.Claim(prodSlice, consSlice);
+        var slip = _processBuilder.Build();
+
+        // Assert
+        slip.Should().NotBeNull();
+
+        var (t1, a1) = slip.Itinerary[0].ShouldBeTransactionWithEvent<AllocatedEvent>(
```
Comment (tnickelsen) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/RegistryProcessBuilder/ClaimTests.cs:64]: I assume you are testing the correct activities are spawned?

File: src/ProjectOrigin.WalletSystem.IntegrationTests/FlowTests/ClaimTests.cs (hunk 1 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/FlowTests/ClaimTests.cs:48]: To test both slices are sliced and then claimed the requested 150, the wallet allows partial claiming, it doesn't force claiming everything one has.

File: src/ProjectOrigin.WalletSystem.Server/RegistryProcessBuilder/Slice.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            Quantity = newSecretCommitmentInfo.Message,
+            RandomR = newSecretCommitmentInfo.BlindingValue.ToArray(),
+            SliceState = SliceState.Registering
+        };
+
+        await _unitOfWork.CertificateRepository.InsertSlice(newSlice);
+        return newSlice;
+    }
+
+    private void BuildSliceRoutingSlip(DepositEndpoint remainderEndpoint, Slice sourceSlice, IHDPrivateKey privateKey, params Slice[] newSlices)
```
[the remark this answers is not in the document]
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.Server/RegistryProcessBuilder/Slice.cs:56]: https://masstransit.io/documentation/patterns/routing-slip

**Review by tnickelsen — APPROVED** [review]

**Pull request merged**